src/v/archival/ntp_archiver_service.cc

```diff
@@ -150,12 +174,81 @@ ss::future<> ntp_archiver::upload_loop() {
     }
 }
 
+ss::future<> ntp_archiver::sync_manifest_loop() {
+    while (sync_manifest_loop_can_continue()) {
+        cloud_storage::download_result result = co_await sync_manifest();
+
+        if (result != cloud_storage::download_result::success) {
+            // The logic in class `remote` already does retries: if we get here,
+            // it means the download failed after several retries, indicating
+            // something non-transient may be wrong. Hence error severity.
+            vlog(
+              _rtclog.error,
+              "Failed to download manifest {}",
+              _manifest.get_manifest_path());
+        } else {
+            vlog(
+              _rtclog.debug,
+              "Successfuly downloaded manifest {}",
+              _manifest.get_manifest_path());
+        }
+
+        if (!sync_manifest_loop_can_continue()) {
+            break;
+        }
+    }
+}
+
+ss::future<cloud_storage::download_result> ntp_archiver::sync_manifest() {
+    retry_chain_node fib(
+      _manifest_upload_timeout, _cloud_storage_initial_backoff, &_rtcnode);
+    retry_chain_logger ctxlog(archival_log, fib, _ntp.path());
+
+    cloud_storage::download_result r = co_await download_manifest();
+    if (r == cloud_storage::download_result::success) {
+        vlog(ctxlog.debug, "Downloading manifest in read-replica mode");
+        if (_partition->archival_meta_stm()) {
+            vlog(
+              ctxlog.debug,
+              "Updating the archival_meta_stm in read-replica mode");
+            auto deadline = ss::lowres_clock::now() + _manifest_upload_timeout;
+            auto error = co_await _partition->archival_meta_stm()->add_segments(
+              _manifest, deadline, _as);
+            if (
+              error != cluster::errc::success
+              && error != cluster::errc::not_leader) {
+                vlog(
+                  ctxlog.warn,
+                  "archival metadata STM update failed: {}",
+                  error);
+            }
+            auto last_offset
+              = _partition->archival_meta_stm()->manifest().get_last_offset();
+            vlog(ctxlog.debug, "manifest last_offset: {}", last_offset);
+        }
+    } else {
+        vlog(
+          ctxlog.error,
+          "Failed to download partition manifest in read-replica mode");
+    }
+
+    co_await ss::sleep_abortable(_sync_manifest_timeout, _as);
```

Comment:
nit: if the sleep here for the benefit of the loop above, then it would be more idiomatic to introduce the sleep in the loop.

Reply:
done